src/drizzleReactHooks.js

```diff
@@ -0,0 +1,203 @@
+              ][cacheKey].value
+          }
+        }
+      }, args)
+      return isFunction
+        ? methodNameOrFunction((contractName, methodName, ...args) => {
+            const cacheKey = drizzle.contracts[contractName].methods[
+              methodName
+            ].cacheCall(...args)
+            return (
+              drizzleState[contractName][methodName][cacheKey] &&
+              drizzleState[contractName][methodName][cacheKey].value
+            )
+          })
+        : drizzleState.value
+    },
+    [drizzle.contracts]
+  )
+  const useCacheSend = useCallback(
+    (contractName, methodName) => {
+      const drizzleState = useDrizzleState(drizzleState => ({
+        transactionStack: drizzleState.transactionStack,
+        transactions: drizzleState.transactions
+      }))
+      const [stackIDs, setStackIDs] = useState([])
+      const transactions = stackIDs.map(
+        stackID =>
+          drizzleState.transactions[
+            drizzleState.transactionStack[stackID] || 'undefined'
+          ]
+      )
+      return {
+        send: (...args) =>
+          setStackIDs(stackIDs => [
+            ...stackIDs,
+            drizzle.contracts[contractName].methods[methodName].cacheSend(
+              ...args
+            )
+          ]),
+        status:
+          transactions[transactions.length - 1] &&
+          transactions[transactions.length - 1].status,
+        transactions
+      }
+    },
+    [drizzle.contracts]
+  )
+  const useCacheEvents = useCallback(
+    (contractName, eventName, eventOptions) => {
+      const [events, setEvents] = useState()
+      const contract = useMemo(
+        () =>
+          new drizzle.web3.eth.Contract(
+            drizzle.contracts[contractName].abi,
+            drizzle.contracts[contractName].address
+          ),
+        [contractName]
+      )
+      useEffect(
+        () => {
+          let mounted = true
+          contract
+            .getPastEvents(eventName, eventOptions)
+            .then(pastEvents => mounted && setEvents(pastEvents))
+          const listener = drizzle.contracts[contractName].events[eventName]({
+            ...eventOptions,
+            fromBlock: 'latest'
+          }).on('data', event => setEvents(events => [...events, event]))
+          return () => {
+            listener.unsubscribe()
+            mounted = false
+          }
+        },
+        [contractName, eventName, eventOptions]
+      )
+      return events
+    },
+    [drizzle.web3, drizzle.contracts]
+  )
+  return (
+    <Context.Provider
+      value={useMemo(
+        () => ({
+          drizzle,
+          useCacheCall,
+          useCacheEvents,
+          useCacheSend
+        }),
+        [drizzle, useCacheCall, useCacheEvents, useCacheSend]
+      )}
+    >
+      {children}
+    </Context.Provider>
+  )
+}
+
+DrizzleProvider.propTypes = {
+  children: PropTypes.node.isRequired,
+  drizzle: PropTypes.shape({}).isRequired
+}
+
+export const Initializer = ({
```

Comment:
Nice work on the initializer, I like the render prop options you gave for each loading state.